src/cmd/troubleshoot/check_image_pullable.go

```diff
@@ -0,0 +1,375 @@
+
+	imageWorks := false
+
+	// parse docker config
+	var result Auths
+	json.Unmarshal([]byte(pullSecret), &result)
+
+	for registry, endpoint := range result.Auths {
+		tslog.Infof("checking images for registry '%s'", registry)
+
+		apiToken := base64.StdEncoding.EncodeToString([]byte(endpoint.Username + ":" + endpoint.Password))
+
+		if statusCode, err := connectToDockerRegistry(httpClient, "HEAD", "https://"+registry+"/v2/", "Basic", apiToken); err != nil {
+			tslog.Errorf("registry '%s' unreachable", registry)
+			continue
+		} else {
+			if statusCode != 200 {
+				tslog.Errorf("registry '%s' unreachable (%d)", registry, statusCode)
+				continue
+			} else {
+				tslog.Infof("registry '%s' is accessible", registry)
+			}
+		}
+
+		if statusCode, err := connectToDockerRegistry(httpClient, "HEAD", "https://"+registry+"/v2/"+componentImage+"/manifests/"+componentVersion, "Basic", apiToken); err != nil {
+			tslog.Errorf("registry '%s' unreachable", registry)
+			continue
+		} else {
+			if statusCode != 200 {
+				tslog.Errorf("image '%s' with version '%s' not found on registry '%s'", componentImage, componentVersion, registry)
+				continue
+			} else {
+				tslog.Infof("image '%s' with version '%s' exists on registry '%s", componentImage, componentVersion, registry)
+			}
+		}
+
+		imageWorks = true
+	}
+
+	if imageWorks {
+		tslog.Okf("%s image '%s' found", componentName, componentRegistry+"/"+componentImage)
+	} else {
+		tslog.Errorf("%s image '%s' missing", componentName, componentRegistry+"/"+componentImage)
+		return fmt.Errorf("%s image '%s' missing", componentName, componentRegistry+"/"+componentImage)
+	}
+	return nil
+}
+
+func addProxy(apiReader client.Reader, troubleshootContext *TestData, httpClient *http.Client) error {
+	dynakube := dynatracev1beta1.DynaKube{}
+	if err := apiReader.Get(context.TODO(), client.ObjectKey{Name: troubleshootContext.dynakubeName, Namespace: troubleshootContext.namespaceName}, &dynakube); err != nil {
+		tslog.WithErrorf(err, "selected '%s:%s' Dynakube does not exist", troubleshootContext.namespaceName, troubleshootContext.dynakubeName)
+		return err
+	}
+
+	proxyUrl := ""
+	if dynakube.Spec.Proxy.Value != "" {
+		proxyUrl = dynakube.Spec.Proxy.Value
+	} else if dynakube.Spec.Proxy.ValueFrom != "" {
+		proxySecret := corev1.Secret{}
+		if err := apiReader.Get(context.TODO(), client.ObjectKey{Name: dynakube.Spec.Proxy.ValueFrom, Namespace: troubleshootContext.namespaceName}, &proxySecret); err != nil {
+			tslog.WithErrorf(err, "'%s:%s' proxy secret is missing", dynakube.Spec.Proxy.ValueFrom, troubleshootContext.dynatraceApiSecretName)
+			return err
+		}
+		var err error
+		proxyUrl, err = kubeobjects.ExtractToken(&proxySecret, dtclient.CustomProxySecretKey)
+		if err != nil {
+			tslog.WithErrorf(err, "failed to extract proxy secret field: %w", err)
+			return fmt.Errorf("failed to extract proxy secret field: %w", err)
+		}
+	}
```

Comment:
I believe there is already a function to get the proxy value either from the spec or the secret in the `kubeobjects` package